titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/env/DefaultAggregatingContainerEnvFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.master.mesos.kubeapiserver.direct.env;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.common.runtime.TitusRuntime;
+
+@Singleton
+public class DefaultAggregatingContainerEnvFactory extends AggregatingContainerEnvFactory {
+
+    private static final String CONFLICT_COUNTER = "titus.aggregatingContainerEnv.conflict";
+
+    private final Registry registry;
+
+    private final Id conflictId;
+
+    @Inject
+    public DefaultAggregatingContainerEnvFactory(TitusRuntime titusRuntime) {
+        super(UserProvidedContainerEnvFactory.getInstance(), TitusProvidedContainerEnvFactory.getInstance());
```

Comment:
It seems like we're depending on the order here to make sure that factories with higher precedence override others. If that's the case, maybe we should add a comment here or where we do the put()/overwrite?

Reply:
Good point.